docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+By: `CS2113T-T13-2`      Since: `Feb 2020`    
+<small>[Go to Webpage](https://ay1920s2-cs2113t-t13-2.github.io/tp/DeveloperGuide.html)</small>
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## **Table of Contents**  
 
-## User Stories
+<big style="color: green">**Introduction** [&#10149;](#introduction)  </big>  
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Purpose [&#10149;](#purpose)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Scope [&#10149;](#scope)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Design Goals [&#10149;](#design-goals)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Definitions [&#10149;](#definitions)   
+<big style="color: green"> **Setting Up** [&#10149;](#setting-up)  </big>  
+<big style="color: green">  **Design** [&#10149;](#design)  </big>  
+<big style="color: green"> **Implementation** [&#10149;](#implementation)  </big>  
+&nbsp; &nbsp; &nbsp; &nbsp; 1. Add Feature [&#10149;](#1-add-feature)   
+&nbsp; &nbsp; &nbsp; &nbsp; 2. Delete Feature [&#10149;](#2-delete-feature)   
+<big style="color: green"> **Appendix** [&#10149;](#appendix)  </big>  
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; User Stories [&#10149;](#user-stories)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Non-Functional Requirements [&#10149;](#non-functional-requirements)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Glossary [&#10149;](#glossary)   
+&nbsp; &nbsp; &nbsp; &nbsp; &#8226; Manual Testing [&#10149;](#manual-testing)    
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+<br>  
 
-## Non-Functional Requirements
+## **Introduction**  
 
-{Give non-functional requirements}
+### **Purpose**  
+<span style="text-align: justify; display: block">
+This document describes the structure and software design decisions for the <b>Nuke</b> application. The <b>Nuke</b> application is a simple yet powerful task management application that is dedicated to providing <b>NUS students</b> a more efficient way to organise their <i>modules</i> and <i>tasks</i>.  
+</span>
+    
 
-## Glossary
+### **Scope**  
+<span style="text-align: justify; display: block">
+This document will cover the structure and software design decisions for the implementation of <b>Nuke</b>. The intended audience for this document are developers, designers and software testers of <b>Nuke</b> <i>or</i> other similar task management application.
+</span>  
 
-* *glossary item* - Definition
+### **Design Goals**  
 
-## Instructions for Manual Testing
+```
+	// To be done.
+```
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### **Definitions**  
+```
+	// To be done.
+```
+
+<br>
+
+## **Setting Up**  
+Refer to the guide [here](#) to set up.  
+
+<br>
+
```

Comment:
Maybe adding a high level architecture view at the beginning would be helpful. I think it would've made it easier for me to read the implementation stuff if I had some more context

Reply:
i know the developer guide is unfinished but just to keep in mind